app/services/encryption/encryptors/small_aes_encryptor.rb

```diff
@@ -0,0 +1,36 @@
+module Encryption
+  module Encryptors
+    class SmallAesEncryptor
+      def initialize
+        self.cipher = SmallAesCipher.new
+      end
+
+      def encrypt(plaintext, cek)
+        payload = fingerprint_and_concat(plaintext)
+        cipher.encrypt(payload, cek)
+      end
+
+      def decrypt(ciphertext, cek)
+        decrypt_and_test_payload(ciphertext, cek)
+      end
+
+      private
+
+      attr_accessor :cipher
+
+      def fingerprint_and_concat(plaintext)
+        fingerprint = Pii::Fingerprinter.fingerprint(plaintext)
```

Comment:
`PII::FIngerprinter.fingerprint` returns a hexdigested fingerprint. We could save even more space using bytes here

Reply:
Oh, that's a good catch! I tried this a bit, and the improvements are relatively constant. It's 20-32 bytes saved regardless of the size of the total payload, so the savings may not necessarily scale. I think the only place it is used is once in the AES encryption, which would track with that.

244c4c063 is my draft attempt at this, I'm kind of on the fence on the tradeoff. If it's useful for the future, it would be more worthwhile though.